RedBeanPHP/QueryWriter/AQueryWriter.php

```diff
@@ -906,42 +906,72 @@ public function updateRecord( $type, $updatevalues, $id = NULL )
 	/**
 	 * @see QueryWriter::parseJoin
 	 */
-	public function parseJoin( $type, $sql, $overrideLinkType = FALSE )
+	public function parseJoin( $type, $sql )
 	{
 		if ( strpos( $sql, '@' ) === FALSE ) {
 			return $sql;
 		}
-		$aliases = OODBBean::getAliases();
+
 		$sql = ' ' . $sql;
 		$joins = array();
 		$joinSql = '';
-		$criteria = array();
-		$joins = array();
-		$startType = $type;
-		$matches = array();
-		if (!preg_match_all('/@((shared|own|joined)\.\S+)/', $sql, $matches)) return $sql;
-		if (count($matches)<3) return $sql;
+
+		if ( !preg_match_all( '#@((shared|own|joined)\.[^\s(,=!?]+)#', $sql, $matches) )
+			return $sql;
+
+		if ( count( $matches ) < 3 )
```

Comment:
Is there ever a case where we can end up with less than 3 matches?

Reply:
Actually no, you're correct.

I thought we would if there was no match but we will always have 3 matches so we should check if `$matches[1]` is empty instead.